### PR TITLE
HRCPP-315 # Pass failed servers to balancing

### PR DESCRIPTION
**Pull request by rigazilla**
Failover balancing needs the list of failed servers to work correctly. 
Fix for https://issues.jboss.org/browse/HRCPP-315

### CODE REVIEW

File: jni/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java [hunk 1]
```diff
@@ -43,12 +44,13 @@ public Configuration(org.infinispan.client.hotrod.jni.Configuration jniConfigura
...
       this.marshaller = null;
       this.servers = null;
+      this.failoverServers = null;
       this.transportFactory = null;
    }
 
    Configuration(ExecutorFactoryConfiguration asyncExecutorFactory, Class<? extends RequestBalancingStrategy> balancingStrategy, ClassLoader classLoader,
          ConnectionPoolConfiguration connectionPool, int connectionTimeout, Class<? extends ConsistentHash>[] consistentHashImpl, boolean forceReturnValues, int keySizeEstimate, Class<? extends Marshaller> marshallerClass,
-         boolean pingOnStartup, String protocolVersion, List<ServerConfiguration> servers, int socketTimeout, SslConfiguration ssl, boolean tcpNoDelay,
+         boolean pingOnStartup, String protocolVersion, List<ServerConfiguration> servers, List<ServerConfiguration> failOverervers, int socketTimeout, SslConfiguration ssl, boolean tcpNoDelay,
```
Comment (tristantarrant) [inline jni/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java]: s/failOverervers/failoverServers/

File: jni/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java [hunk 2]
```diff
@@ -60,6 +62,7 @@ public Configuration(org.infinispan.client.hotrod.jni.Configuration jniConfigura
       this.marshallerClass = marshallerClass;
       this.marshaller = null;
       this.servers = Collections.unmodifiableList(servers);
+      this.failoverServers = Collections.unmodifiableList(servers);
```
Comment (tristantarrant) [inline jni/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java]: Why are you passing in failoverServers, but then assigning servers here ?

Comment (rigazilla) [inline jni/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java]: Wrong merge. Fixed thanks

**Comment by tristantarrant** [comment]
Merged to master, thanks

**Pull request closed without merging**